pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1103,12 +1103,13 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
-// GetAllocatableDevices returns information about all the devices known to the manager
+// GetAllocatableDevices returns information about all the healthy devices known to the manager
 func (m *ManagerImpl) GetAllocatableDevices() ResourceDeviceInstances {
 	m.mutex.Lock()
-	resp := m.allDevices.Clone()
-	m.mutex.Unlock()
-	klog.V(4).InfoS("Known devices", "numDevices", len(resp))
+	defer m.mutex.Unlock()
+	klog.V(4).InfoS("Known devices", "numDevices", len(m.allDevices))
```

Comment:
Do we really need / want this printed here?  Presumably the caller will only be interested in the "allocatable" devices (since they are calling a function called `GetAllocatableDevices()`). I don't feel too strongly since this is a `V(4)` log, but it seems unintuitive to me to include it here.

Reply:
having both the information about all and filtered out devices helps in troubleshooting, but I believe a good middleground could be coalescing the logs in one, keeping the `V(4)`.